src/main/js/tansaction/transaction-modal.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Bank account modal component.
+ *
+ * @author Xiaobing Hou
+ */
+import React from "react";
+import { Modal } from "react-bootstrap";
+import TransactionForm from "./transaction-form";
+
+const TransactionModal = ({
+  bankAccountId,
+  isAddTransaction,
+  selectedTransaction,
+  handleEditTransactions,
+  handleAddTransactions,
+  ...props
+}) => {
+  return (
+    <Modal
+      {...props}
+      size="md"
+      aria-labelledby="contained-modal-title-vcenter"
+      centered
+    >
+      <Modal.Header closeButton>
+        <Modal.Title id="contained-modal-title-vcenter">
+          Bank account Id: {bankAccountId}
+          <br />
+          Transaction Id:{" "}
+          {selectedTransaction
+            ? isAddTransaction
+              ? "None"
+              : selectedTransaction.id
+            : "None"}
```

Comment:
I have an example with accounts which handle this cleaner without passing a flag down.